components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/impl/SubscriptionsApiServiceImplTestCase.java

```diff
@@ -0,0 +1,192 @@
+import java.util.UUID;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(RestApiUtil.class)
+public class SubscriptionsApiServiceImplTestCase {
+
+    private static final String USER = "admin";
+    private static final String contentType = "application/json";
+
+    @Test
+    public void testSubscriptionsGetfromAPI() throws APIManagementException, NotFoundException {
+        TestUtil.printTestMethodName();
+        String apiId = UUID.randomUUID().toString();
+        String subsID1 = UUID.randomUUID().toString();
+        String subsID2 = UUID.randomUUID().toString();
+
+        SubscriptionsApiServiceImpl subscriptionsApiService = new SubscriptionsApiServiceImpl();
+        APIStore apiStore = Mockito.mock(APIStoreImpl.class);
+
+        PowerMockito.mockStatic(RestApiUtil.class);
+        PowerMockito.when(RestApiUtil.getConsumer(USER)).thenReturn(apiStore);
+        PowerMockito.when(RestApiUtil.getLoggedInUsername()).thenReturn(USER);
+
+        List<Subscription> subscriptionList = new ArrayList<>();
+        subscriptionList.add(SampleTestObjectCreator.createSubscription(subsID1));
+        subscriptionList.add(SampleTestObjectCreator.createSubscription(subsID2));
+
+        Mockito.when(apiStore.getSubscriptionsByAPI(apiId)).thenReturn(subscriptionList);
+
+        Response response = subscriptionsApiService.subscriptionsGet
+                (apiId, null, null, 0, 10, contentType, null, TestUtil.getRequest());
+
+        Assert.assertEquals(200, response.getStatus());
+    }
+
+    @Test
+    public void testSubscriptionsGetfromApplication() throws APIManagementException, NotFoundException {
+        TestUtil.printTestMethodName();
+        String applicationId = UUID.randomUUID().toString();
+        String subsID1 = UUID.randomUUID().toString();
+        String subsID2 = UUID.randomUUID().toString();
+
+        SubscriptionsApiServiceImpl subscriptionsApiService = new SubscriptionsApiServiceImpl();
+        APIStore apiStore = Mockito.mock(APIStoreImpl.class);
+
+        PowerMockito.mockStatic(RestApiUtil.class);
+        PowerMockito.when(RestApiUtil.getConsumer(USER)).thenReturn(apiStore);
+        PowerMockito.when(RestApiUtil.getLoggedInUsername()).thenReturn(USER);
+
+        Application application = SampleTestObjectCreator.createDefaultApplication();
+        List<Subscription> subscriptionList = new ArrayList<>();
+        subscriptionList.add(SampleTestObjectCreator.createSubscription(subsID1));
+        subscriptionList.add(SampleTestObjectCreator.createSubscription(subsID2));
+
+        Mockito.when(apiStore.getApplicationByUuid(applicationId)).thenReturn(application);
+        Mockito.when(apiStore.getAPISubscriptionsByApplication(application, ApiType.STANDARD))
+                .thenReturn(subscriptionList);
+
+        Response response = subscriptionsApiService.subscriptionsGet
+                (null, applicationId, ApiType.STANDARD.toString(), 0, 10, contentType, null, TestUtil.getRequest());
```

Comment:
contentType variable is removed due to Store API refactoring. I feel this will lead to a compilation issue.